tests/pathResolver.test.js

```diff
@@ -197,6 +197,24 @@ describe('pathResolver.js', () => {
 					}
 				}
 			},
+			{
+				name: 'dynamic value with url encoded component',
+				path: `/resource/${encodeURIComponent('resourceId/path:&-/')}`,
+				inputMap: {
+					resource: {
+						'*': {
+							_value: expectedValue,
+							_tokens: ['token1']
+						}
+					}
+				},
+				expectedValue: {
+					value: expectedValue,
+					tokens: {
+						token1: 'resourceId/path:&-/'
```

Comment:
Why is the token not `path:&-` like in the other test cases? (I'm likely missing the use case of this change, so this might actually be desired, but from looking at it, it isn't straight forward to understand.)

Reply:
It's about automatically url decoding a path parameter. So ignore the `/` for now, assume your resource is `/resources/{resourceId}` and someone passes `resource|part2` as the path parameter. Do you want `request.pathParameters.resourceId` to be `resource%7Cpart2` or `resource|part2`.

This uriDecodes the `resourceId` automatically without needing to do that in the controller method.

Reply:
Agree. However, I'd assume that only the relevant parts are decoded:

If we have `/resources/{resourceId}`, for `/resources/1`, I'd expect `token1: '1'` and if it's `/resources/something|encoded` then I'd expect `token1: 'something|encoded'`, but it seems to be `token1: '/resources/something|encoded'`. More or less, I'd expect `/resources/` to be the "variable" of the token1 parameter, at least when looking at the other unit tests.

Reply:
Only the relevant parts are decoded... I think the value I used to test that decoding works correctly is a bit confusing, I'll change it to something else.

Reply:
I pulled the value into a variable, hopefully that makes it clearer what is happening.